tests/integration_tests.rs

```diff
@@ -649,6 +701,39 @@ fn simple_for_merger_loop() {
     unsafe { weld_value_free(ret_value) };
 }
 
+fn simple_zipped_for_merger_loop() {
+    #[allow(dead_code)]
+    struct Args {
+        x: WeldVec<i32>,
+        y: WeldVec<i32>,
+    }
+
+    let code = "|x:vec[i32], y:vec[i32]| result(for(zip(x,y), merger[i32,+], |b,i,e| merge(b, e.$0+e.$1)))";
+    let conf = default_conf();
+
+    let size = 2000;
+    let x_data = vec![1; size as usize];
+    let y_data = vec![1; size as usize];
+
+    let ref input_data = Args {
+        x: WeldVec {
+            data: x_data.as_ptr(),
+            len: x_data.len() as i64,
+        },
+        y: WeldVec {
+            data: x_data.as_ptr(),
+            len: x_data.len() as i64,
```

Comment:
shouldn't this be y_data? I get an "unused variable" on y_data, although it is identical to x_data in this case